tests/Contao/WidgetTest.php

```diff
@@ -66,6 +67,27 @@ public function testGetPost($key, $input, $value, $expected)
         error_reporting($errorReporting);
     }
 
+    public function testValidate()
+    {
+        $widget = $this->getMockBuilder('Contao\Widget')
+            ->disableOriginalConstructor()
+            ->setMethods(['validator', 'generate'])
+            ->getMock();
+
+        $widget->expects($this->exactly(2))
+            ->method('validator')
+            ->withAnyParameters()
+            ->willReturnArgument(0);
+
+        /** @var Widget $widget */
+        $widget->validate('foobar');
+        $this->assertSame('foobar', $widget->value);
+
+        /** @var Widget $widget */
+        $widget->validate(null, true);
+        $this->assertNull($widget->value);
+    }
```

Comment:
Right, there was even a bug which I fixed with the unit test covering that case :-D See df901dfa58b38a67c82a4c39753342294183cf93.